### ISSUE TITLE
Problem with generating urls deeper than level 4 in the categories

### ISSUE THREAD

**Issue by celocore** [olegkoval/magento2-regenerate_url_rewrites#47]
Hello,

we have a Magento shop (version 2.2.6, PHP 7.1), with 5 category levels according to the default category. The products are in the levels 3 to 5. If we use the extension (version 1.2.3) to regenerate the rewrite urls, the last level of the product urls is always deleted and the product url key is attached. The urls for the categories are generated correctly. However, this does not occur until level 4.
A resulting problem is that there are double urls for products. In some cases the urls of products and categories are identical due to the shortening. So they are not created and the error "URL key for specified store already exists. acknowledges the problem.

Does anyone know the problem and has a solution?

Catalog Structure:

```
Default Category
|-> Level 1
|-> Level 1
    |-> Level 2
        |-> Level 3
        |-> Level 3
            |-> Level 4
                |-> Level 5
            |-> Level 4
            |-> Level 4
```

Product URL in Level 2: https://www.domain.tld/level-1/level-2/product-url-key.html
Product URL in **Level 3**: https://www.domain.tld/level-1/level-2/level-3/product-url-key.html
Product URL in **Level 4**: https://www.domain.tld/level-1/level-2/level-3/product-url-key.html
Product URL in Level 5: https://www.domain.tld/level-1/level-2/level-3//level-4/product-url-key.html

kind regards
Michael

**Comment by olegkoval** [comment]
Hello @celocore 

Please, update extension to the version [1.3.0](https://github.com/olegkoval/magento2-regenerate_url_rewrites/releases/tag/v1.3.0) and check if issue still exists in your store.

I've tested in my sandbox and everything works Ok.

**Comment by celocore** [comment]
Hello @olegkoval 

I updated the extension to version 1.3.0. Unfortunately the problem still exists. Max depth of Request Path Level is 3 with the same behavior as in the first post.

kind regards
Michael

**Comment by tuyennn** [comment]
Confirmed this was fixed on latest v1.3.1 - I have 6 level of categories and all work as expected

**Comment by olegkoval** [comment]
Fixed in v1.3.1. Closed.
